### ISSUE TITLE
get_reliable_rw() broken in Release mode

### ISSUE THREAD

**Issue by coolstar** [0x36/oob_events#4]
get_reliable_rw() is broken in Release mode and fails to find a suitable IOSurface offset to write into

Disabling clang optimization on this function resolves the issue

**Comment by 0x36** [comment]
Hi, thanks for your feedback, I suspect you are talking about get_target_iosurface(), get_reliable_rw() has nothing to do with IOSurface object writes.

**Comment by coolstar** [comment]
no, I'm talking about get_reliable_rw(). Adding the `__attribute__ ((optnone))` macro to it works around the issue in Release builds (but of course it disables compiler optimizations in that function)

**Comment by coolstar** [comment]
oops yeah I said the part it fails at wrong. It actually fails at "Overwriting shmem size to have safe OOB"

**Comment by 0x36** [comment]
hmm, I barely notice this issue on my devices, but you can however get rid of the whole function if you want, with some slight modification in get_target_iosurface() it'd work well and may also improve the reliability on devices with higher memory size.
